app/controllers/api/base_controller/renderer.rb

```diff
@@ -166,7 +166,9 @@ def virtual_attribute_search(resource, attribute)
         if resource.class < ApplicationRecord
           # is relation in 'attribute' variable plural in the model class (from 'resource.class') ?
           if [:has_many, :has_and_belongs_to_many].include?(resource.class.reflection_with_virtual(attribute).try(:macro))
-            Rbac.filtered(resource.public_send(attribute))
+            resource_attr = resource.public_send(attribute)
+            return resource_attr unless resource_attr.first.kind_of?(ApplicationRecord)
```

Comment:
it is under `if` with condition  `is it a plural relation?` so I think it has to be guaranteed. Or rather add `try(:first)`?